### PR TITLE
Cleanup: Rename the provider-level fields to increase clarity.

### PR DESCRIPTION
**Pull request by mattmoor**
:broom: This renames the provider-level fields to qualify them in ways that better convey their semantics contributing to the `apko_config`.

🚨 THIS IS A BREAKING CHANGE 🚨 

/kind cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,14 +14,14 @@ This provides an `apko_build` resource that will build the provided `apko` confi
 # with additional packages.
 provider "apko" {
   # Default to building for these architectures.
-  archs = ["x86_64", "aarch64"]
+  default_archs = ["x86_64", "aarch64"]
```
Comment (imjasonh) [inline README.md:17]: IIRC this overwrites a config's archs if set, but we could double-check that with a test.

Comment (mattmoor) [reply README.md:17]: Hmm, yeah lemme check that.

Comment (mattmoor) [reply README.md:17]: Ok, I updated this with the new semantic and a test for that.

File: internal/provider/config_data_source_test.go [hunk 2]
```diff
@@ -128,6 +128,47 @@ contents:
...
+				archs:        []string{"x86_64", "aarch64"},
+				packages:     []string{"wolfi-baselayout"},
+			}),
+		},
+		Steps: []resource.TestStep{{
+			Config: `
+data "apko_config" "this" {
+  config_contents = <<EOF
+archs:
+  - aarch64
```
Comment (mattmoor) [inline internal/provider/config_data_source_test.go:147]: cc @imjasonh FYI

**Review by imjasonh — APPROVED** [review]

**Pull request merged**